### PR TITLE
Updates to test code to support JDK 21 containers.

### PR DESCRIPTION
**Pull request by sefroberg**
The testcode updates needed to support the Red Hat OpenJDK 21 containers.

### CODE REVIEW

File: test-openjdk/pom.xml [hunk 1]
```diff
@@ -231,12 +237,7 @@
             </activation>
             <dependencies>
                 <dependency>
-                    <!--Need tools.jar in compilation classpath. JDK 1.7 no longer has it there by default: -->
-                    <groupId>com.sun</groupId>
-                    <artifactId>tools</artifactId>
-                    <version>${maven.compiler.target}</version>
-                    <scope>system</scope>
-                    <systemPath>${java.home}/../lib/tools.jar</systemPath>
+
```
Comment (judovana) [inline test-openjdk/pom.xml:240]: I htink this remans valid for jdk8

File: test-openjdk/src/test/java/com/redhat/qe/openjdk/maven/MavenJavaArgsTest.java [hunk 2]
```diff
@@ -180,7 +180,7 @@ public void javaClassPathTest() {
 	@Test
 	public void javaDiagnosticsTest() {
 		deploy(JavaS2IBuild.DOCKER_IMAGE_TEST_APP.getManagedBuild(), Collections.singletonMap("JAVA_DIAGNOSTICS", "true"));
-		if (OpenJDKTestConfig.isOpenJDK17()) {
+		if ( OpenJDKTestConfig.isOpenJDK21() ||OpenJDKTestConfig.isOpenJDK17()) {
```
Comment (judovana) [inline test-openjdk/src/test/java/com/redhat/qe/openjdk/maven/MavenJavaArgsTest.java:183]: why there are configs/ifs for 17 and 21, an not for 11 and 8?

Comment (sefroberg) [reply test-openjdk/src/test/java/com/redhat/qe/openjdk/maven/MavenJavaArgsTest.java:183]: JDK 11 is captured in line 187 and then jdk 8 line 191.

File: test-openjdk/pom.xml (hunk 1 above)
Comment (sefroberg) [reply test-openjdk/pom.xml:240]: Hmm, I did not notice any issues with ubi 8 jdk8. I will try rhel 7 jdk8 to make sure that does not break too.

Comment (judovana) [reply test-openjdk/pom.xml:240]: fair enogh then. If you tried, its ok. I hav enoticed several issues with tools.jar over years, so was doubel checking it. No need to rerun in el7, the packaging is same

File: test-openjdk/src/test/java/com/redhat/qe/openjdk/maven/MavenJavaArgsTest.java (hunk 2 above)
Comment (judovana) [reply test-openjdk/src/test/java/com/redhat/qe/openjdk/maven/MavenJavaArgsTest.java:183]: ty!

**Review by judovana — APPROVED** [review]
ok

File: test-openjdk/src/test/java/com/redhat/qe/openjdk/image/AbstractDockerImageTest.java [hunk 3]
```diff
@@ -57,7 +57,14 @@ public abstract class AbstractDockerImageTest extends OpenJDKTestParent {
 			"jps", "jrunscript", "jshell", "jstack", "jstat",
 			"jstatd", "keytool", "rmiregistry", "serialver"
 	};
-
+	public static final String[] DEFAULT_JAVA_21_UTILITIES = new String[]{
+			"alt-java", "jarsigner", "javac", "javap", "jconsole",
```
Comment (jmtd) [inline test-openjdk/src/test/java/com/redhat/qe/openjdk/image/AbstractDockerImageTest.java:61]: I see alt-java in the runtime images (=headless non-devel RPM) but not the builders, oddly

Comment (jmtd) [reply test-openjdk/src/test/java/com/redhat/qe/openjdk/image/AbstractDockerImageTest.java:61]: Let me expand on that. In the builder images, alt-java is in /usr/bin but not in any of /usr/lib/jvm/*/bin. I haven't traced in the test where this list of binaries is checked against. My entire comment may be a waste of time

Comment (jmtd) [reply test-openjdk/src/test/java/com/redhat/qe/openjdk/image/AbstractDockerImageTest.java:61]: also FWIW jcmd is now in the runtime images, and didn't used to be

**Review by jmtd — APPROVED** [review]

**Comment by judovana** [comment]
Two nits:
jcmd was very likely moved in all jdks, except 8. Nice catch!

@jmtd  Ca y ou please elaborate more on that h alt-java thing? IIRC, it hsoudl be in all headles jdks.

File: test-openjdk/src/test/java/com/redhat/qe/openjdk/image/AbstractDockerImageTest.java (hunk 3 above)
Comment (sefroberg) [reply test-openjdk/src/test/java/com/redhat/qe/openjdk/image/AbstractDockerImageTest.java:61]: This testsuite currently does not run against the runtimes due to the historic nature of the tests.

**Comment by jmtd** [comment]
> @jmtd Ca y ou please elaborate more on that h alt-java thing? IIRC, it hsoudl be in all headles jdks.

It's in the headless RPM for sure: it's just not in /usr/lib/jvm:

```shell
[jboss@4b53aa94af56 ~]$ rpm -qa java-*
java-21-openjdk-headless-21.0.0.0.35-2.el8.x86_64
java-21-openjdk-devel-21.0.0.0.35-2.el8.x86_64
java-21-openjdk-21.0.0.0.35-2.el8.x86_64
[jboss@4b53aa94af56 ~]$ ls /usr/lib/jvm/*/bin/alt-java
ls: cannot access '/usr/lib/jvm/*/bin/alt-java': No such file or directory
[jboss@4b53aa94af56 ~]$ command -v alt-java
/usr/bin/alt-java
[jboss@4b53aa94af56 ~]$ readlink -f $(command -v alt-java)
/usr/bin/alt-java-21
[jboss@4b53aa94af56 ~]$ rpm -qf $(!!)
rpm -qf $(readlink -f $(command -v alt-java))
java-21-openjdk-headless-21.0.0.0.35-2.el8.x86_64
[jboss@4b53aa94af56 ~]$ find /usr/lib/jvm -name alt-java
[jboss@4b53aa94af56 ~]$ 
```

**Comment by sefroberg** [comment]
OK, @judovana @jmtd If you both can take one more look at this please.

**Comment by judovana** [comment]
Tahnx for asking. If there is somethign missing, I can not say.  Current patch looks good.

**Review by andrlos — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -9,8 +9,10 @@ This test suite is designed to be executed against current versions of Red Hat's
 | UBI 8                    | 8               | [openjdk-8](https://catalog.redhat.com/software/containers/ubi8/openjdk-8/5dd6a48dbed8bd164a09589a)                                  |
 | UBI 8                    | 11              | [openjdk-11](https://catalog.redhat.com/software/containers/ubi8/openjdk-11/5dd6a4b45a13461646f677f4)                                |
 | UBI 8                    | 17              | [openjdk-17](https://catalog.redhat.com/software/containers/ubi8/openjdk-17/618bdbf34ae3739687568813)                                |
+| UBI 8                    | 21              | TBD                                                                                                                                  |
```
Comment (jmtd) [inline README.md:12]: we've since shipped these images so you could fill out the table with the JDK21 image URIs now. Or do it in a future commit

**Review by jmtd — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (sefroberg) [reply README.md:12]: Future commit. At this point I just want to get this code in so we are ready for the CPU.

**Comment by sefroberg** [comment]
Running tests on this today. Assuming that all passes I will merge this.

**Pull request merged**